src/libsyntax/ast.rs

```diff
@@ -1927,6 +1927,10 @@ pub enum ItemKind {
     ///
     /// E.g. `trait Foo { .. }` or `trait Foo<T> { .. }`
     Trait(Unsafety, Generics, TyParamBounds, Vec<TraitItem>),
+    /// Trait alias
+    /// 
+    /// E.g. `trait Foo = Bar + Quux;`
+    TraitAlias(Generics, TyParamBounds),
```

Comment:
I need to reread the RFC, but IIRC there should be two sets of predicates - one for the alias itself (LHS) and one for the alias substitution result.
I.e.
```rust
trait Alias<T: Clone> = where T: Default;

=>

trait Alias<T> whereLHS T: Clone = whereRHS T: Default;
```
(But maybe they are equivalent?)

Reply:
I mean `Alias<NoClone>` is an immediate error, but `Alias<CloneNoDefault>` is a valid always-false predicate.

Reply:
I think they are equivalent  - both are added as predicates.

Reply:
This should be related to https://github.com/rust-lang/rust/issues/21903 somehow.

Reply:
The RFC only mentions `whereRHS`.

Reply:
@durka 
Could you prohibit bounds on trait alias' generic parameters in `ast_validation` for now?
```
trait TraitAlias<T: Bound /* <-- error, this one represents "whereLHS" */>
```

Reply:
Sounds good.

Reply:
Added an error for this and also `trait Alias<T = ()>`. [cfail](https://github.com/rust-lang/rust/pull/45047/files#diff-c764b8ef26325201c0a34cadf64a856a)